draft-zsf.md

```diff
@@ -0,0 +1,167 @@
+
+- **Unmined ZEC is now accounted for as `ZSF_BALANCE`**
+- **Transaction includes optional contributions to ZSF via a `ZSF_DEPOSIT` field**
+- Thus, at Every New Block:
+    - Miner and funding streams rewarded the same constant amount, **but from `ZSF_BALANCE`** (this constant amount still halves at specified heights)
+    - Miner is rewarded via Transaction fees `(inputs - outputs)`, **where `outputs` includes the `ZSF_DEPOSIT` amount**
+
+For example, an end-user wallet application could have an option to contribute a portion of a transaction to the ZSF, which would be included in a `ZSF_DEPOSIT` field in the transaction, to be taken into account by the Zcash nodes.
+
+This quite simple alteration has -- in our opinion -- a multitude of benefits:
+
+1. **Long Term Consensus Sustainability:** This mechanism supports long-term consensus sustainability by addressing concerns about the sustainability of the network design shared by Bitcoin-like systems through the establishment of deposits into the Sustainability Fund to augment block rewards, ensuring long-term sustainability as the issuance rate of Zcash drops and newly issued ZEC decreases over time.
+2. **Benefits to ZEC Holders:** Deposits to the ZSF slow down the payout of ZEC, temporarily reducing its available supply, benefiting current holders of unencumbered active ZEC in proportion to their holdings without requiring them to opt into any scheme, introducing extra risk, active oversight, or accounting complexity.
+3. **Recovery of "Soft-Burned" Funds:** In some instances, such as miners not claim all available rewards, some ZEC goes unaccounted for though not formally burned. This proposal would recover it through the `ZSF_BALANCE` mechanism described below.
+
+# Specification
+
+In practice, The Zcash Sustainability Fund is a single global balance maintained by the node state and contributed to via a single transaction field. This provides the economic and security support described in the motivation section, while also importantly keeping the fund payouts extremely simple to describe and implement.
+
+The two modifications are:
+1. The re-accounting of unmined ZEC as a node state field called `ZSF_BALANCE`
+2. The addition of a transaction field called `ZSF_DEPOSIT`
+
+## `ZSF_BALANCE`
+
+The ZEC issuance mechanism is re-defined to remove funds from `ZSF_BALANCE`, which is initially set to `MAX_MONEY` at the genesis block.
+
+Consensus nodes are then required to track new per-block state:
+
+- `ZSF_BALANCE[H] : u64 [zatoshi]`
+
+The state is a single 64 bit integer (representing units of `zatoshi`) at any given block height, ``H``, representing the Sustainability Fund balance at that height, ``H``. The `ZSF_BALANCE` can be calculated using the following formula:
+
+`ZsfBalanceAfter(height) = MAX_MONEY + sum_{h = 0}^{height} (ZsfDeposit(height) + Unclaimed(height) - BlockSubsidy(height))`
```

Comment:
Bug: The sum is ignoring the sum variable and is instead repeatedly counting the last block in the range.
```suggestion
`ZsfBalanceAfter(height) = MAX_MONEY + sum_{h = 0}^{height} (ZsfDeposit(h) + Unclaimed(h) - BlockSubsidy(h))`
```

Reply:
The ZIP Editors had a not-insubstantial discussion about off-by-one errors, and decided that it would be useful to define `ZsfBalanceDelta(height)` as the change in the ZSF balance as a result of applying the block at `height` to the chain consensus state. Then other values can be derived from this in a way that more closely matches how full nodes will actually implement it:
```
ZsfBalanceBefore(height) = MAX_MONEY if height == 0 else ZsfBalanceAfter(height - 1)
ZsfBalanceAfter(height) = ZsfBalanceBefore(height) + ZsfBalanceDelta(height)
```